tests/actions/IOUTest.ts

```diff
@@ -6299,6 +6299,60 @@ describe('actions/IOU', () => {
                 });
             });
         });
+
+        it('should remove all existing category violations when the transaction "Category" is unset', async () => {
```

Comment:
This unit test is flaky and is not passing on `main`. I've tried re-running it twice and it's failed each time.

<img width="1341" height="548" alt="Image" src="https://github.com/user-attachments/assets/6d75b19a-96c4-47b6-b9ba-07c8740b3912" />

Example run: https://github.com/Expensify/App/actions/runs/20240609444/job/58117837402

@dmkt9 Can you please look into this?